bot/exts/evergreen/trivia_quiz.py

```diff
@@ -1,56 +1,234 @@
+    "tesla": ("(unit of magnetic flux density)", "kg*s^-2*A^-1"),
+}
+
+
+def linear_system(q_format: str, a_format: str) -> QAndA:
+    """Generate a system of linear equations with two unknowns."""
+    x, y = random.randint(2, 5), random.randint(2, 5)
+    answer = a_format.format(x, y)
+
+    coeffs = random.sample(range(1, 6), 4)
+
+    question = q_format.format(
+        coeffs[0],
+        coeffs[1],
+        (
+            coeffs[0] * x + coeffs[1] * y
+        ),
+        coeffs[2],
+        coeffs[3],
+        (
+            coeffs[2] * x + coeffs[3] * y
+        ),
+    )
+
+    return question, answer
+
+
+def mod_arith(q_format: str, a_format: str) -> QAndA:
+    """Generate a basic modular arithmetic question."""
+    quotient, m, b = random.randint(30, 40), random.randint(10, 20), random.randint(200, 350)
+    ans = random.randint(0, 9)  # max 9 because min mod 10
+    a = quotient * m + ans - b
+
+    question = q_format.format(a, b, m)
+    answer = a_format.format(ans)
+
+    return question, answer
+
+
+def ngonal_prism(q_format: str, a_format: str) -> QAndA:
+    """Generate a question regarding vertices on n-gonal prisms."""
+    n = random.randint(0, len(N_PREFIXES) - 1)
+
+    question = q_format.format(N_PREFIXES[n])
+    answer = a_format.format((n + N_PREFIX_STARTS_AT) * 2)
+
+    return question, answer
+
+
+def imag_sqrt(q_format: str, a_format: str) -> QAndA:
+    """Generate a negative square root question."""
+    ans_coeff = random.randint(3, 10)
+
+    question = q_format.format(ans_coeff ** 2)
+    answer = a_format.format(ans_coeff)
+
+    return question, answer
+
+
+def binary_calc(q_format: str, a_format: str) -> QAndA:
+    """Generate a binary calculation question."""
+    a = random.randint(15, 20)
+    b = random.randint(10, a)
+    oper = random.choice(
+        (
+            ("+", operator.add),
+            ("-", operator.sub),
+            ("*", operator.mul),
+        )
+    )
+
+    if oper[0] == "*":
+        a -= 5
+        b -= 5
+
+    question = q_format.format(
+        a,
+        oper[0],
+        b,
+    )
+    answer = a_format.format(oper[1](a, b))
+
+    return question, answer
+
+
+def solar_system(q_format: str, a_format: str) -> QAndA:
+    """Generate a question on the planets of the Solar System."""
+    planet = random.choice(PLANETS)
+
+    question = q_format.format(planet[0])
+    answer = a_format.format(planet[1])
+
+    return question, answer
+
+
+def taxonomic_rank(q_format: str, a_format: str) -> QAndA:
+    """Generate a question on taxonomic classification."""
+    level = random.randint(0, len(TAXONOMIC_HIERARCHY) - 2)
+
+    question = q_format.format(TAXONOMIC_HIERARCHY[level])
+    answer = a_format.format(TAXONOMIC_HIERARCHY[level + 1])
+
+    return question, answer
+
+
+def base_units_convert(q_format: str, a_format: str) -> QAndA:
+    """Generate a SI base units conversion question."""
+    unit = random.choice(list(UNITS_TO_BASE_UNITS.keys()))
```

Comment:
```suggestion
    unit = random.choice(list(UNITS_TO_BASE_UNITS))
```

Reply:
oh yes, that was redundant. resolved

Reply:
the `.keys()` version is more clear to me, but I guess that's subjective